internal/pkg/aws/codestar/codestar_test.go

```diff
@@ -0,0 +1,78 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package codestar
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/codestarconnections"
+
+	"github.com/aws/copilot-cli/internal/pkg/aws/codestar/mocks"
+
+	"github.com/golang/mock/gomock"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestCodestar_WaitUntilStatusAvailable(t *testing.T) {
+	t.Run("times out if connection status not changed to available in allotted time", func(t *testing.T) {
+		// GIVEN
+		ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(time.Second))
```

Comment:
nit: can we do:
```suggestion
		ctx, cancel := context.WithDeadline(context.Background(), time.Now())
```
so that the test fails immediately instead of taking  a second.